src/features/search/store.ts

```diff
@@ -0,0 +1,46 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { SearchResult } from './components/types';
+import { remoteList, RemoteList } from 'utils/storeUtils';
+
+export type SearchResultItem = {
+  id: string;
+  result: SearchResult;
+};
+
+export interface SearchStoreSlice {
+  matchesByQuery: Record<string, RemoteList<SearchResultItem>>;
+}
+
+const initialState: SearchStoreSlice = {
+  matchesByQuery: {},
+};
+
+const searchSlice = createSlice({
+  initialState,
+  name: 'search',
+  reducers: {
+    resultsError: (state, action: PayloadAction<[string, unknown]>) => {
+      const [query, error] = action.payload;
+      state.matchesByQuery[query] = remoteList();
+      state.matchesByQuery[query].error = error;
```

Comment:
nice! I have not seen the error part of these in action anywhere else i think, seems like something we want to take into more parts!

Reply:
True! I just delayed error handling (lazily) but there was a test here that failed, and I had to decide whether to rip out the test and all the error code, or just implement crude error handling, so I chose the latter. 😊